### PR TITLE
Fix: Add attachmentType to envelope ser/deser.

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Fix: Add attachmentType to envelope ser/deser.

## :bulb: Motivation and Context
Extract from https://github.com/getsentry/sentry-java/pull/1256/files

## :pencil: Checklist

- [X] I reviewed the submitted code
- [X] I added tests to verify the changes
- [ ] I updated the docs if needed
- [X] No breaking changes

### CODE REVIEW

File: sentry/src/main/java/io/sentry/Attachment.java [hunk 1]
```diff
@@ -140,6 +146,33 @@ public Attachment(
     this.addToTransactions = addToTransactions;
   }
 
+  /**
+   * Initializes an Attachment with a path, a filename, a content type, and addToTransactions.
```
Comment (philipphofmann) [inline sentry/src/main/java/io/sentry/Attachment.java:150]: `l`:
```suggestion
   * Initializes an Attachment with a path, a filename, a content type, addToTransactions, and attachmentType.
```

File: sentry/src/main/java/io/sentry/SentryEnvelopeItemHeader.java [hunk 2]
```diff
@@ -42,23 +43,36 @@ public int getLength() {
       final @NotNull SentryItemType type,
       int length,
       final @Nullable String contentType,
-      final @Nullable String fileName) {
+      final @Nullable String fileName,
+      final @Nullable String attachmentType) {
```
Comment (philipphofmann) [inline sentry/src/main/java/io/sentry/SentryEnvelopeItemHeader.java:47]: `m`: I see that we often pass `null` for `attachmentType`. Does it maybe make sense to define an overload so we don't need to pass `null`?

**Review by philipphofmann — APPROVED** [review]
Just two minor comments. TIL. Didn't know the abbreviation ser/deser.

File: sentry/src/main/java/io/sentry/SentryEnvelopeItemHeader.java (hunk 2 above)
Comment (marandaneto) [reply sentry/src/main/java/io/sentry/SentryEnvelopeItemHeader.java:47]: makes sense

**Review by Swatinem — APPROVED** [review]

**Pull request merged**